homeassistant/components/holiday/calendar.py

```diff
@@ -0,0 +1,129 @@
+"""Holiday Calendar."""
+from __future__ import annotations
+
+from datetime import datetime
+
+from holidays import country_holidays
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_PROVINCE, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Holiday Calendar config entry."""
+    country: str = config_entry.data[CONF_COUNTRY]
+    province: str | None = config_entry.data.get(CONF_PROVINCE)
+
+    async_add_entities(
+        [
+            HolidayCalendarEntity(
+                hass,
+                config_entry.title,
+                country,
+                province,
+                config_entry.entry_id,
+            )
+        ],
+        True,
+    )
+
+
+class HolidayCalendarEntity(CalendarEntity):
+    """Representation of a Holiday Calendar element."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        country: str,
+        province: str | None,
+        unique_id: str,
+    ) -> None:
+        """Initialize HolidayCalendarEntity."""
+        self._country = country
+        self._province = province
+        self._location = name
+        self._event: CalendarEvent | None = None
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name=name,
+        )
+
+        obj_holidays = country_holidays(self._country, subdiv=self._province)
+        available_languages = [
+            lang.replace("en_US", "en") for lang in obj_holidays.supported_languages
+        ]
+
+        if hass.config.language in available_languages:
+            self._default_language = hass.config.language
+        else:
+            self._default_language = obj_holidays.default_language or "en_US"
+
+        if self._default_language == "en" and self._country != "CA":
+            self._default_language = "en_US"
```

Comment:
Can we do this in `async_setup_entry()` instead before setting up the entity and passing in the right language

Reply:
Done.